### DISCUSSION TITLE
The column `active` does not exist in the current database.

### DISCUSSION THREAD

**Discussion by arpitdalal** [prisma/prisma#21379] [Q&A] [answered]
### Question

I've following prisma schema:
```prisma
model User {
  id       String  @id @default(cuid())
  email    String  @unique
  username String  @unique
  name     String?
  active   Boolean @default(true)

  createdAt DateTime @default(now())
  updatedAt DateTime @updatedAt

  password    Password?
  sessions    Session[]
}

model Session {
  id             String   @id @default(cuid())
  expirationDate DateTime

  createdAt DateTime @default(now())
  updatedAt DateTime @updatedAt

  user   User   @relation(fields: [userId], references: [id], onDelete: Cascade, onUpdate: Cascade)
  userId String

  // non-unique foreign key
  @@index([userId])
}
```

Everything is working fine in the app, but in one of my vitest tests I do following:
```ts
const session = await prisma.session.create({
  data: {
	  expirationDate: getSessionExpirationDate(),
	  user: {
		  create: {
			  username: "john",
                          name: "John Doe",
                          email: "john@example.com"
		  },
	  },
  },
  select: {
	  id: true,
	  userId: true,
  },
})
```

And the test is failing saying `The column "active" does not exist in the current database.`
But the `active` columns do exist in the database, I've seen it using prisma studio. But for some reason, this query thinks that there is not active column 🤷‍♂️ And I'm not even trying to set any value to the `active` column, it defaults to true.

I've migrated every migrations, and as I said before, I've seen the `active` column in the prisma studio so it is there.

### Expected behavior (optional)

The expected behavior is that this should work if I'm not missing anything.

### Information about Prisma Schema, Client Queries and Environment (optional)

- OS: macOS
- Database: SQLite
- Node.js version: 18.17.1

- Run `prisma -v` to see your Prisma version and paste it
```sh
prisma                  : 5.3.1
@prisma/client          : 5.3.1
```

**Comment by janpio** [comment]
Prisma Studio shows you the fields from your model, not necessarily what exists in your database. Does it also show data for that column? Can you confirm the column exists via a different database GUI?

**Reply by arpitdalal** [reply]
What other gui can I use on my local? Can I run any commands to make sure that my db is up to date to my schema?

**Reply by janpio** [reply]
`prisma db push`, or if you are using migrations, `migrate deploy` (or `migrate dev` if you have potentially not created all migrations yet).

I locally use DBeaver. I hear TablePlus is good on MacOS.

**Reply by arpitdalal** [reply]
I used https://sqliteviewer.app to view my local db file as I'm using sqlite and here's that column:
<img width="978" alt="image" src="https://github.com/prisma/prisma/assets/61059807/8086ba92-d59a-465c-b0be-53f333dcbaa0">

**Reply by arpitdalal** [reply]
I also tried `prisma db push` and received this
```sh
╰─ npx prisma db push                                                                                                    ─╯
Environment variables loaded from .env
Prisma schema loaded from prisma/schema.prisma
Datasource "db": SQLite database "data.db" at "file:./data.db?connection_limit=1"

The database is already in sync with the Prisma schema.

✔ Generated Prisma Client (v5.3.1) to ./node_modules/@prisma/client in 123ms
```

**Reply by arpitdalal** [reply]
and here're the results for
`prisma migrate dev`
```sh
╰─ npx prisma migrate dev                                                                                                 ─╯
Environment variables loaded from .env
Prisma schema loaded from prisma/schema.prisma
Datasource "db": SQLite database "data.db" at "file:./data.db?connection_limit=1"

Already in sync, no schema change or pending migration was found.

✔ Generated Prisma Client (v5.3.1) to ./node_modules/@prisma/client in 147ms
```
and `prisma migrate deploy`
```sh
╰─ npx prisma migrate deploy                                                                                              ─╯
Environment variables loaded from .env
Prisma schema loaded from prisma/schema.prisma
Datasource "db": SQLite database "data.db" at "file:./data.db?connection_limit=1"

16 migrations found in prisma/migrations


No pending migrations to apply.
```

**Reply by janpio** [reply]
Thanks for double checking.

I quickly reproduced your situation with the information you provided, ran `db push`, and then `node script.js` - and it works perfectly fine: https://github.com/janpio/21379

So something has to be wrong with the database that is being used during your test run.

**Reply by arpitdalal** [reply, marked as answer]
Thanks for checking it. It was my bad 🤦‍♂️ tests were using a different db than the app and needed to migrate that db that was used by tests. Thanks very much for helping me and sorry to totally waste your time 😅

**Reply by janpio** [reply]
All good, only thing counts is that you figured it out. I also learned about https://sqliteviewer.app/, which is actually buy a person that I know in a totally different context 😆

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
